### ISSUE TITLE
hc with Credentials?

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#2360]
### Discussed in https://github.com/orgs/honojs/discussions/2291

<div type='discussions-op-text'>

<sup>Originally posted by **lajbel** March  1, 2024</sup>
Looks pretty obvious to have something like `credentials` in the configuration of `hc`, something like this

```js
const authClient = hc<SomeType>("/api/auth/", {
    headers: {} // You can already set headers!
    credentials: "include",
});
```

I was searching here and there's only one option available to do this approach

```js
const api = hc<AppType>("api/auth", {
  fetch: (req, init) =>
    fetch(req, {
      ...init,
      credentials: "include", // Why not have all fetch options too?
    })
})
```

RPC is pretty awesome, but some things are missing</div>

**Comment by yusukebe** [comment]
You can see more details in #2291.

**Comment by yusukebe** [comment]
Fixed by #2592
